services/apps/portal/src/application/application.controller.ts

```diff
@@ -49,14 +52,37 @@ export class ApplicationController {
   }
 
   @Post('/:fileId')
-  async update(@Req() req, @Param('fileId') fileId: string) {
+  async update(
+    @Param('fileId') fileId: string,
+    @Body() updateDto: UpdateApplicationDto,
+    @Req() req,
+  ) {
+    //Verify User has Access
+    const existingApplication = this.applicationService.getByFileId(
+      fileId,
+      req.user.entity,
+    );
+
+    if (!existingApplication) {
+      throw new Error('Failed to find application with given File ID and User');
```

Comment:
Maybe ServiceNotFoundException instead of Error?

Reply:
Done!